src/Analyzers/CSharp/Analyzers/RemoveRedundantElseStatement/RemoveRedundantElseStatementDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,136 @@
+    {
+        public RemoveRedundantElseStatementDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.RemoveRedundantElseStatementDiagnosticId,
+                   EnforceOnBuildValues.RemoveRedundantElseStatement,
+                   CSharpCodeStyleOptions.PreferRemoveRedundantElseStatement,
+                   new LocalizableResourceString(nameof(CSharpAnalyzersResources.Remove_redundant_else_statement), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(context =>
+            {
+                context.RegisterSyntaxNodeAction(AnalyzeSyntax, SyntaxKind.IfStatement);
+            });
+        }
+
+        private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var codeStyleOption = context
+                .GetCSharpAnalyzerOptions()
+                .PreferRemoveRedundantElseStatement;
+
+            if (!codeStyleOption.Value)
+                return;
+
+            var ifStatement = (IfStatementSyntax)context.Node;
+
+            if (ifStatement.Parent is ElseClauseSyntax)
+            {
+                // Only offer on top most if
+                return;
+            }
+
+            var redundantElse = FindRedundantElse(ifStatement);
+            if (redundantElse is not null)
+            {
+                context.ReportDiagnostic(DiagnosticHelper.Create(
+                    Descriptor,
+                    redundantElse.ElseKeyword.GetLocation(),
+                    codeStyleOption.Notification.Severity,
+                    additionalLocations: ImmutableArray.Create(redundantElse.GetLocation()),
+                    properties: null));
+            }
+        }
+
+        private static ElseClauseSyntax? FindRedundantElse(IfStatementSyntax ifStatement)
+        {
+            var elseClause = ifStatement.Else;
+
+            while (elseClause is not null)
+            {
+                var endsWithJump = ifStatement.Statement switch
+                {
+                    BlockSyntax block => IsJumpStatement(block.Statements.LastOrDefault()),
+                    _ => IsJumpStatement(ifStatement.Statement),
+                };
+
+                // doing this only makes sense when every if ends with a jump
+                if (!endsWithJump)
+                {
+                    return null;
+                }
+
+                if (elseClause.Statement is IfStatementSyntax elseIfStatement)
+                {
+                    elseClause = elseIfStatement.Else;
+                }
+
+                // reached else not followed by an if
+                break;
```

Comment:
i don't understand this.  you always hit this, even if you hit an if-statement above.  did you miss a continue?